src/locomotion_mode.cpp

```diff
@@ -0,0 +1,375 @@
+#include "locomotion_mode/locomotion_mode.hpp"
+
+#include <regex>
+
+namespace locomotion_mode{
+
+LocomotionMode::LocomotionMode(rclcpp::NodeOptions options, const std::string node_name)
+: Node(node_name,
+    options.allow_undeclared_parameters(true).
+    automatically_declare_parameters_from_overrides(true)),
+  // Protected
+  node_name_(node_name),
+  enabled_(false)
+{
+  // Load Parameters
+  load_params();
+
+  // Load URDF
+  if(!load_robot_model()){
+    RCLCPP_ERROR(this->get_logger(), "Loading of robot model failed!");
+  }
+
+  // Create Services
+  enable_service_ = this->create_service<std_srvs::srv::Trigger>(
+    "~/enable",
+    std::bind(
+      &LocomotionMode::enable_callback, this, std::placeholders::_1,
+      std::placeholders::_2));
+  disable_service_ = this->create_service<std_srvs::srv::Trigger>(
+    "~/disable",
+    std::bind(
+      &LocomotionMode::disable_callback, this, std::placeholders::_1,
+      std::placeholders::_2));
+
+  // Create Publishers
+  joint_command_publisher_ = this->create_publisher<rover_msgs::msg::JointCommandArray>(
+    "joint_cmds", 10);
+
+  // Create Subscriptions
+  joint_state_subscription_ = this->create_subscription<sensor_msgs::msg::JointState>(
+    "joint_states", 10, std::bind(
+      &LocomotionMode::joint_state_callback, this,
+      std::placeholders::_1));
+
+  RCLCPP_INFO(this->get_logger(), "%s STARTED.", node_name.c_str());
+}
+
+// Dummy Callback function in case the derived class forgets to create a custom callback function
+void LocomotionMode::rover_velocities_callback(const geometry_msgs::msg::Twist::SharedPtr msg)
+{
+  rover_msgs::msg::JointCommandArray joint_command_array_msg;
+
+  joint_command_array_msg = compute_joint_commands(msg);
+
+  // TODO: Check size of array
+  // Publish Message
+  joint_command_publisher_->publish(joint_command_array_msg);
+}
+
+rover_msgs::msg::JointCommandArray LocomotionMode::compute_joint_commands(
```

Comment:
```suggestion
// Compute the rover joint commands based on the body velocity. Should be overwritten.
rover_msgs::msg::JointCommandArray LocomotionMode::compute_joint_commands(
```

Add comment to method